src/main/java/sign_up/interface_adapters/SignUpViewModel.java

```diff
@@ -0,0 +1,35 @@
+package sign_up.interface_adapters;
+
+import sign_up.use_case.*;
+
+public class SignUpViewModel implements SignUpOutputBoundary {
+    // The view which will be updated following the viewmodel
+    private final View view;
+    private final SignUpInputBoundary inputBoundary;
+
+    public SignUpViewModel(SignUpDataAccessInterface dataAccess, View view) {
+        inputBoundary = new SignUpInteractor(dataAccess, this);
```

Comment:
Is this acting as both the presenter and the controller? Might be better to split into different classes (probably less likely to violate Clean Architecture). As it is now, I think this might be violating the Single Responsibility Principle.

Reply:
I had the same concern, but I've talked to the professors about this at office hours and they say that it is okay in this case being the use of the MVVM model

Reply:
Looking at the way you've implemented the Controller and Presenter with you VM, is this how MVVM is supposed to work? I can implement in a similar way if needed.